File: test/novel/novel.py
import dataclasses

@dataclasses.dataclass
class Book:
    title:    str
    left_page: int
    size:     int

    def turn_page(self, count: int):
        self.left_page += 2 * count

        if self.left_page < 0:
            self.left_page = 0
        elif self.left_page >= self.size:
            self.left_page = self.size - 1

    def get_page(self) -> int:
        return self.left_page

File: test/novel/novel.js
exports.Book = class {
  constructor(prop) {
    this.title = prop.title;
    this.leftPage = prop.left_page;
    this.size = prop.size;
  }

  turnPage(count) {
    this.leftPage += 2 * count;

    if (this.leftPage < 0) {
      this.leftPage = 0;
    } else if (this.leftPage > this.size) {
      this.leftPage = this.size - 1;
    }
  }

  getPage() {
    return this.leftPage;
  }
};
